Packs/ApiModules/Scripts/EWSApiModule/EWSApiModule.py

```diff
@@ -0,0 +1,718 @@
+            try:
+                config_args = get_config_args_from_context(context_dict, self.credentials)
+                account = Account(
+                    primary_smtp_address=target_mailbox,
+                    autodiscover=False,
+                    config=Configuration(**config_args),
+                    access_type=self.access_type,
+                    default_timezone=time_zone
+                )
+                account.root.effective_rights.read  # noqa: B018 pylint: disable=E1101
+                return account
+            except Exception as e:
+                # fixing flake8 correction where original_exc is assigned but unused
+                original_exc = e
+
+        try:
+            account = Account(
+                primary_smtp_address=self.account_email,
+                autodiscover=True,
+                credentials=self.credentials,
+                access_type=self.access_type,
+            )
+        except AutoDiscoverFailed:
+            return_error('Auto discovery failed. Check credentials or configure manually')
+
+        new_context = cache_autodiscover_results(context_dict, account)
+        if new_context == context_dict and original_exc:
+            # Autodiscovery returned the same connection params as the cached ones we failed to use
+            raise original_exc  # pylint: disable=E0702
+
+        if target_mailbox == self.account_email:
+            demisto.setIntegrationContext(new_context)
+
+        return account
+
+    def get_items_from_mailbox(self, account: Optional[Union[Account, str]], item_ids):
+        """
+        Request specific items from a mailbox associated with an account
+
+        :param account: EWS account or target_mailbox associated with that account
+        :param item_ids: item_ids of the requested items
+
+        :return: list of exchangelib Items
+        """
+        # allow user to pass target_mailbox as account
+        if not isinstance(account, Account):
+            account = self.get_account(account) if isinstance(account, str) else self.get_account(self.account_email)
+
+        if not isinstance(item_ids, list):
+            item_ids = [item_ids]
+
+        items = [Item(id=x) for x in item_ids]
+        result = list(account.fetch(ids=items))
+        result = [x for x in result if not (isinstance(x, ErrorItemNotFound | ErrorInvalidIdMalformed))]
+        if len(result) != len(item_ids):
+            raise Exception('One or more items were not found/malformed. Check the input item ids')
+        return result
+
+    def get_item_from_mailbox(self, account: Optional[Union[Account, str]], item_id):
+        """
+        Request a single item from a mailbox associated with an account
+
+        :param account: EWS account or target_mailbox associated with that account
+        :param item_id: item_id of the requested item
+
+        :return: exchangelib Item
+        """
+        result = self.get_items_from_mailbox(account, [item_id])
```

Comment:
Why is this get item[s] if we send a single item? 
Also inside this function we have an error if the item was not found. 
Lets say we get the error then in what case the validation and if len(result) == 0 relevant here?